src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/PremisesService.kt

```diff
@@ -91,4 +95,74 @@ class PremisesService(
 
       return success(lostBedsEntity)
     }
+
+  fun createNewPremises(requestBody: NewPremises) = validated<PremisesEntity> {
```

Comment:
Elsewhere we've avoided passing the generated-from-OpenAPI request models into the service layer so that if we need another external interface later (e.g. an MQ) the service layer is agnostic to that.  So I'd perhaps just pass down the individual properties you need as top level instead.

Thinking about it as I wrote this, there is at least one area we've fallen foul of this already (the JSON Path keys when building a validation response being the example that comes to mind.)

Reply:
Oh okay, so should I still just pass down the properties needed and not the request model?

Reply:
Yeah that's right - bit of pedantry I suppose but may as well be consistent :)

Reply:
Now done